src/core/utils/slaac_address.cpp

```diff
@@ -219,7 +261,27 @@ void Slaac::AddAddresses(void)
     {
         bool found = false;
 
-        if (!ShouldUseForSlaac(config))
+        if (!IsSlaac(config) || IsFiltered(config))
+        {
+            continue;
+        }
+
+        for (SlaacAddress &slaacAddr : mSlaacAddresses)
+        {
+            if (slaacAddr.IsInUse() && DoesConfigMatchNetifAddr(config, slaacAddr))
+            {
+                if (slaacAddr.IsDeprecating())
+                {
+                    slaacAddr.MarkAsNotDeprecating();
+                    Get<ThreadNetif>().UpdatePreferredFlagOn(slaacAddr, true);
```

Comment:
The `true` parameter here is not correct for all cases. If the newly added prefix uses P_preferred = false in the Border Router TLV, then the SLAAC address must also have P_preferred = false.   This is an uncommon case, but still, it could happen.   So an address can be at the same time "deprecated" and "not deprecating".

Reply:
Thanks for noticing this. Yes. That's correct.  Changed the code to check the prefix flag from netdata  -> `if (slaacAddr.IsDeprecating() && config.mPreferred)`